File: backend/src/Logitar.Portal.Contracts/Configurations/Configuration.cs
using Logitar.Portal.Contracts.Settings;

namespace Logitar.Portal.Contracts.Configurations;

public class Configuration : Aggregate
{
  public Locale? DefaultLocale { get; set; }
  public string Secret { get; set; }

  public UniqueNameSettings UniqueNameSettings { get; set; } = new();
  public PasswordSettings PasswordSettings { get; set; } = new();
  public bool RequireUniqueEmail { get; set; }

  public LoggingSettings LoggingSettings { get; set; } = new();

  public Configuration() : this(string.Empty)
  {
  }

  public Configuration(string secret)
  {
    Secret = secret;
  }
}

File: frontend/src/types/configuration.ts
import type { Aggregate } from "@/types/aggregate";
import type { Locale } from "@/types/i18n";
import type { PasswordSettings, UniqueNameSettings } from "@/types/settings";

export type Configuration = Aggregate & {
  defaultLocale?: Locale;
  secret: string;
  uniqueNameSettings: UniqueNameSettings;
  passwordSettings: PasswordSettings;
  requireUniqueEmail: boolean;
  loggingSettings: LoggingSettings;
};

export type LoggingExtent = "ActivityOnly" | "Full" | "None";

export type LoggingSettings = {
  extent: LoggingExtent;
  onlyErrors: boolean;
};

export type ReplaceConfigurationPayload = {
  defaultLocale?: string;
  secret: string;
  uniqueNameSettings: UniqueNameSettings;
  passwordSettings: PasswordSettings;
  requireUniqueEmail: boolean;
  loggingSettings: LoggingSettings;
};
